### PR TITLE
use features

### PR DESCRIPTION
**Pull request by Dillon-Roller**

### CODE REVIEW

File: client/Cargo.toml [hunk 1]
```diff
@@ -11,36 +11,39 @@ build_common = { path = "../build_common" }
...
-dioxus-fullstack = { git = "https://github.com/OxideOps/dioxus.git", features = ["desktop"] }
-dioxus-desktop = { git = "https://github.com/OxideOps/dioxus.git" }
-num_cpus = "1.16.0"
-server-functions = { path = "../server-functions", features = ["desktop"] }
-sys-info = "0.9.1"
-tokio-tungstenite = { version = "*", features = ["native-tls"] }
+[features]
+desktop = ["async-process", "dioxus-desktop", "num_cpus", "sys-info"]
+web = ["dioxus-web", "js-sys", "wasm-bindgen", "wasm-bindgen-futures"]
+server = ["tokio/full", "tower-http/fs", "tower", "dioxus-fullstack/axum", "axum"]
```
Comment (DanielAlbl) [inline client/Cargo.toml:49]: I think this might have to be called 'ssr' to make the server functions work.

**Review by DanielAlbl — APPROVED** [review]

File: client/src/client/system_info/web.rs [hunk 2]
```diff
@@ -3,6 +3,7 @@ pub(crate) fn get_num_cores() -> usize {
     js_result.as_f64().unwrap_or(1.0) as usize
 }
 
+#[cfg(feature = "desktop")]
```
Comment (DanielAlbl) [inline client/src/client/system_info/web.rs:6]: This file is already specifically for web. We could remove this function but it might be useful later. Could just do a clippy ignore or something.

File: client/src/client/components/piece.rs [hunk 3]
```diff
@@ -1,7 +1,7 @@
 use async_std::channel::Receiver;
 use dioxus::{html::geometry::ElementPoint, prelude::*};
 
-use crate::shared_states::BoardSize;
+use super::super::shared_states::BoardSize;
```
Comment (DanielAlbl) [inline client/src/client/components/piece.rs:4]: Is `super::super` an indication we should be doing a `use` in mod.rs?

File: README.md [hunk 4]
```diff
@@ -33,7 +33,7 @@ This project is a complete chess platform, allowing users to play chess games, a
 ## Building and Running
 There are two binary packages that can be compiled and ran: `client` and `server`. Execute `cargo [build | run]` with the `-p` (package) flag, followed by the package:
```
Comment (DanielAlbl) [inline README.md:34]: I guess technically this whole section is wrong now, since there's no server package

File: Cargo.toml [hunk 5]
```diff
@@ -2,10 +2,9 @@
 resolver = "2"
 members = [
     "auto-deref",
-    "client",
-    "server",
+    "app",
     "chess",
     "common",
```
Comment (DanielAlbl) [inline Cargo.toml:7]: I think `common` and `build_common` could also be modules in `app`, but this can easily be a separate PR.

Comment (Dillon-Roller) [reply Cargo.toml:7]: yeah Ill make an issue

File: README.md (hunk 4 above)
Comment (Dillon-Roller) [reply README.md:34]: made issue

**Review by DanielAlbl — APPROVED** [review]
